### PR TITLE
feat: add Wheel Of Fortune command

### PR DESCRIPTION
**Pull request by favna**
- chore: i18n stuff
- added new images for the wheel of fortune. Be sure to upload them to the dev server @kyranet
- Made the translation key for when a user doesn't have enough shinies more generic
- marked Icons enum in SlotMachine as const enum
- cleaned up tsdoc in SlotMachine
- made the variable names in SlotMachine more decriptive
- made it so slotmachine deducts the input from the money amount if the user loses
- made it so slotmachine also shows that a user has lost their input
- use more i18n in SlotMachine
- made slotmachine look like wheel of fortune
- refactor: create constant for social folder

<details><summary>Click here for screenshots</summary>

<span>
<a taget="_blank" href="https://user-images.githubusercontent.com/4019718/70397548-7f42b380-1a13-11ea-8b4f-ae681205515a.png"></a>
<a taget="_blank" href="https://user-images.githubusercontent.com/4019718/70397549-823da400-1a13-11ea-9f5e-81d76270c25d.png"></a>
</span>

</details>

### CODE REVIEW

File: src/lib/util/Games/Slotmachine.ts [hunk 1]
```diff
@@ -110,30 +102,40 @@ export class Slotmachine {
...
-		const amount = this.winnings === 0 ? money - this.amount : money + (this.winnings * this.boost);
-		if (amount < 0) throw 'You cannot have negative money.';
+		const amount = this.winnings === 0 ? money - this.amount : money - this.amount + (this.winnings * this.boost);
+		if (amount < 0) throw this.message.language.tget('GAMES_CANNOT_HAVE_NEGATIVE_MONEY');
 		await settings.update(UserSettings.Money, amount);
-		return this.render(rolls);
+		return {
+			attachment: await this.render(rolls),
+			amount
+		};
```
Comment (favna) [inline src/lib/util/Games/Slotmachine.ts]: I'd prefer if this was an array that could be destructured but I couldn't figure out the typing on the command side of things where the destructuring happens. If someone could help here that be great.

File: src/lib/util/Games/WheelOfFortune.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		const spin = Math.floor(Math.random() * this.MULTIPLIERS.length);
+		this.calculate(spin);
+
+		const amount = this.winnings === 0 ? money - this.amount : money - this.amount + (this.winnings * this.boost);
+		if (amount < 0) throw this.message.language.tget('GAMES_CANNOT_HAVE_NEGATIVE_MONEY');
+		await settings.update(UserSettings.Money, amount);
+		return {
+			attachment: await this.render(money, spin, amount),
+			amount
+		};
```
Comment (favna) [inline src/lib/util/Games/WheelOfFortune.ts]: Same as with slots about the array thing

**Comment by favna** [comment]
\<snip> -> Moved to top comment

File: src/lib/util/Games/Slotmachine.ts (hunk 1 above)
Comment (favna) [reply src/lib/util/Games/Slotmachine.ts]: so returned as a tuple

File: src/commands/Games/wheeloffortune.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			description: language => language.tget('COMMAND_WHEELOFFORTUNE_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_WHEELOFFORTUNE_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
+			runIn: ['text'],
+			usage: '<50|100|200|500|1000|2000|5000|10000>'
+		});
+	}
+
+	public async run(message: KlasaMessage, [text]: [string]) {
+		await WheelOfFortune.init();
```
Comment (vladfrangu) [inline src/commands/Games/wheeloffortune.ts]: This doesn't have to be called every message, does it now? If not, please move to the init function

Comment (vladfrangu) [reply src/commands/Games/wheeloffortune.ts]: In fact, you init it in klasaReady...why call it again?

File: src/lib/util/Games/WheelOfFortune.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	public static images = Object.seal<WheelOfFortuneAssets>({
+		LOSE_ICONS: null,
+		WIN_ICONS: null,
+		ARROWS: null,
+		SHINY: null
+	});
+
+	public static async init(): Promise<void> {
+		const [winIcons, loseIcons, arrows, shiny] = await Promise.all([
+			loadImage(join(assetsFolder, 'images', 'social', 'wof-win-icons.png')),
```
Comment (vladfrangu) [inline src/lib/util/Games/WheelOfFortune.ts]: Can't you join the path to the social assets in a constant, and use that?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/commands/Games/wheeloffortune.ts (hunk 3 above)
Comment (favna) [reply src/commands/Games/wheeloffortune.ts]: woops leftover from testing

Comment (favna) [inline src/commands/Games/wheeloffortune.ts]:
```suggestion
```

File: src/lib/util/Games/WheelOfFortune.ts (hunk 4 above)
Comment (favna) [reply src/lib/util/Games/WheelOfFortune.ts]: Ye I suppose so

**Review by kyranet — CHANGES_REQUESTED** [review]
The lack of padding in the embed must be fixed, as colours should be consistent and use the profile's colour (`#202225`) over the embed's.

image

File: src/commands/Games/wheeloffortune.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+export default class extends SkyraCommand {
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['wof'],
+			cooldown: 10,
+			description: language => language.tget('COMMAND_WHEELOFFORTUNE_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_WHEELOFFORTUNE_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
```
Comment (kyranet) [inline src/commands/Games/wheeloffortune.ts]: You need to add `'EMBED_LINKS'` to slotmachine, and `'ATTACH_FILES'` in this one.

File: src/commands/Games/wheeloffortune.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	private buildEmbed(message: KlasaMessage, attachment: MessageAttachment, money: number, amount: number) {
+		return new MessageEmbed()
+			.setColor(getColor(message))
+			.attachFiles([attachment])
+			.setTitle(message.language.tget('COMMAND_WHEELOFFORTUNE_EMBED_TITLE'))
+			.setImage('attachment://wof.png')
+			.addField(`Previous ${Emojis.Shiny}`, money, true)
+			.addField(`New ${Emojis.Shiny}`, amount, true);
```
Comment (kyranet) [inline src/commands/Games/wheeloffortune.ts]: i18n

File: src/commands/Games/slotmachine.ts [hunk 7]
```diff
@@ -23,11 +26,23 @@ export default class extends SkyraCommand {
...
+	}
+
+	private buildEmbed(message: KlasaMessage, attachment: MessageAttachment, money: number, amount: number) {
+		return new MessageEmbed()
+			.setColor(getColor(message))
+			.attachFiles([attachment])
+			.setTitle(message.language.tget('COMMAND_SLOTS_EMBED_TITLE'))
+			.setImage('attachment://slots.png')
+			.addField(`Previous ${Emojis.Shiny}`, money, true)
+			.addField(`New ${Emojis.Shiny}`, amount, true);
```
Comment (kyranet) [inline src/commands/Games/slotmachine.ts]: i18n

File: src/events/klasaReady.ts [hunk 8]
```diff
@@ -11,6 +12,7 @@ export default class extends Event {
 	public async run() {
 		try {
 			await Slotmachine.init().catch(error => this.client.emit(Events.Wtf, error));
+			await WheelOfFortune.init().catch(error => this.client.emit(Events.Wtf, error));
```
Comment (kyranet) [inline src/events/klasaReady.ts]: Those can run with a Promise.all

File: src/lib/util/Games/Slotmachine.ts [hunk 9]
```diff
@@ -110,30 +102,40 @@ export class Slotmachine {
...
+		return [await this.render(rolls, darkTheme), amount] as [Buffer, number];
 	}
 
-	public async render(rolls: readonly Icons[]) {
-		const win = this.winnings > 0;
-		const length = win ? 300 : 150;
+	public async render(rolls: readonly Icons[], darkTheme: boolean) {
+		const playerHasWon = this.winnings > 0;
+		const darkColour = '#202225';
+		const lightColour = '#FFFFFF';
```
Comment (kyranet) [inline src/lib/util/Games/Slotmachine.ts]: Those should go outside this method.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**